### ISSUE TITLE
Hide "XML Document" Tree View for Non-XML Documents

### ISSUE THREAD

**Issue by ionut-botizan** [DotJoshJohnson/vscode-xml#104]
Like the title says, you should only display the explorer tree view when a user opens a XML document. It's a waste of space having it there all the time.
Not sure about it, but I believe this could be achieved using the `onLanguage` activation event:
https://code.visualstudio.com/docs/extensionAPI/activation-events

**Comment by DotJoshJohnson** [comment]
There is currently no official API available to turn the custom view on and off, but I'll play around with it to see if I can make something work. While an event is raised by VS Code when the user "activates" the view, it is a "one-way" street. I can only listen to the event to know when a user opens the view, I can't raise an event to close the view.

I'll do a little research on this and submit an issue to the main VS Code repo if needed.

**Comment by Danieleeee** [comment]
Hi @DotJoshJohnson ,
you can activate this extension only under xml language.
In package.json:
```
"activationEvents": [
        "onLanguage:xml"
    ],
"contributes": {
        "languages": [{
            "id": "xml",
            "aliases": [
                "XML",
                "xml"
            ],
            "extensions": [
                ".xml"
            ]
        }],
 "grammars": [{
            "language": "xml"
        }],
```

So we avoid triggering the extension with unsupported programming languages ​​and consequently saving resources

**Comment by DotJoshJohnson** [comment]
@Danieleeee - Extension activation is handled in the [`activationEvents`](https://code.visualstudio.com/docs/extensionAPI/activation-events) property. I already only activate the extension on a supported language or when a command from this extension is invoked (see https://github.com/DotJoshJohnson/vscode-xml/blob/master/package.json#L124-L132). The issue is, I cannot "deactivate" the extension once it is activated.

**Comment by Danieleeee** [comment]
There is also the problem that the extension, or at least the tree view, is opened at vscode startup:
![schermata 2017-06-16 alle 18 37 53](https://user-images.githubusercontent.com/22749175/27236020-7bf244ee-52c3-11e7-9378-26705e095fd5.png)
Does not mean the file type you opened

**Comment by DotJoshJohnson** [comment]
@Danieleeee - Right, but as I said, I don't seem to have any control over when the custom view is turned on and off.

It looks like the ability to toggle custom views is being worked on (per Microsoft/vscode#27823). I'll keep an eye on the progression of these APIs. Custom views is a new feature, so I'm sure we'll see plenty of movement in this space in the coming weeks/months.

**Comment by Danieleeee** [comment]
Ok, thanks

**Comment by Silvenga** [comment]
@DotJoshJohnson taking a look at the issue you linked, the ability to selectively hide has been merged:

It's implmented like these:
https://github.com/Microsoft/vscode-extension-samples/blob/master/tree-view-sample/package.json#L30
https://github.com/eamodio/vscode-gitlens/blob/master/package.json#L1519

Looks like it a part of this release: https://github.com/Microsoft/vscode/issues/28387

I can write a pull if you want. 😸

**Comment by DotJoshJohnson** [comment]
@Silvenga - Thanks for the heads up! If you want to submit a PR, I'll be more than happy to merge it in. Otherwise, I'll have time Sunday or Monday to make the change.

**Comment by pltrant** [comment]
package.json should be updated to be only display the view on XML documents or if the flag is not set to true:

```
"explorer": [
                  {		                  {
                      "id": "xmlTreeView",		                      "id": "xmlTreeView",
 -                    "name": "XML Document"		 +                    "name": "XML Document",
 +                    "when": "**resourceLangId == 'xml' |** config.xmlTools.treeViewEnabled != false | "
```

**Comment by Yanpas** [comment]
I don't need it even in xml docs, hope to see config that disables it

**Comment by DotJoshJohnson** [comment]
This will be taken care of in the `v2` release.

**Comment by DotJoshJohnson** [comment]
Released in [`v2.0.0-preview.2`](https://github.com/DotJoshJohnson/vscode-xml/releases/tag/v2.0.0-preview.2).
